geomstats/_backend/tensorflow/__init__.py

```diff
@@ -322,10 +328,19 @@ def assignment_by_sum(x, values, indices, axis=0):
     If a list is given, it must have the same length as indices.
     """
     if _is_boolean(indices):
-        indices = [index for index, val in enumerate(indices) if val]
-
+        if ndim(array(indices)) > 1:
+            indices = tf.where(indices)
+        else:
+            if ndim(x) > 1:
```

Comment:
Found the problem which makes hypersphere's exp fail: this if condition is useless. (And even if it was, there lacks an else counterpart). 
It can be removed in `assignment` and `assignment_by_sum`